app/controllers/idv/phone_controller.rb

```diff
@@ -102,7 +102,15 @@ def failure_url(reason)
 
     def async_state_done(async_state)
       form_result = step.async_state_done(async_state)
-      analytics.track_event(Analytics::IDV_PHONE_CONFIRMATION_VENDOR, form_result.to_h)
+      analytics.track_event(
+        Analytics::IDV_PHONE_CONFIRMATION_VENDOR,
+        form_result.to_h.merge(
+          pii_like_keypaths: [
```

Comment:
With how widespread it seems we have innocent logging by attribute name collisions, I wonder if it's worth considering one of the other avenues considered for detecting PII, like common patterns for phone/SSN/etc, or "known" PII values like our [mock profile details](https://github.com/18F/identity-idp/blob/6648d2a2543b488022f642fb4cda7b4e7026fe3a/app/services/doc_auth/mock/result_response_builder.rb#L4-L19)?

Reply:
My sense is that approach trades convenience for completeness? Many tests used that PII but not all of them do? Do we think it would be close enough to check for those attribute values?

Reply:
I don't really feel too strongly one way or the other. For me, the ticket seemed to be framed as something of a safety net for peace of mind, and not necessarily the primary safeguard against logging PII. I could also wonder if using the mock details could catch logging in a situation where it's not cleanly structured data with attribute names, or with slight variations to the exact attribute name.

The current implementation isn't really problematic to me, though I would wonder if it could be confusing that the `pii_like_keypaths` is not actually part of the "extra" payload being logged.

We could even do both if we wanted 🤷

Reply:
Added in 77a2aad2d1989e7e6e03e89d8a4223e043a88d7b

Example error:
```
FakeAnalytics::PiiDetected: track_event example PII first_name (FAKEY) detected in attributes
event: Trackable Event ()
full event: {:some_benign_key=>"FAKEY MCFAKERSON"}"
```

Reply:
Another option, is we could update `FormResponse` and the classes similar to it to have a new attribute (next to `extra`), which get included in the `.to_h`? like this:

```ruby
FormResponse.new(
  success: false,
  errors: { ssn: 'is not formatted right' },
  extra: { num_attempts: 14 },
  pii_like_key_paths: [[:errors, :ssn], [:error_details, :ssn]],
)
```